File: account-service/src/main/java/com/wks/servicemarketplace/accountservice/core/models/CountryCode.java
package com.wks.servicemarketplace.accountservice.core.models;

import com.wks.servicemarketplace.accountservice.core.exceptions.CountryCodeNotFoundException;
import lombok.NonNull;
import lombok.Value;

import java.util.Optional;

@Value
public class CountryCode {

    @NonNull
    private final String countryCode;

    public CountryCode(String countryCode) throws CountryCodeNotFoundException {
        this.countryCode = Optional.ofNullable(com.neovisionaries.i18n.CountryCode.getByAlpha2Code(countryCode))
                .map(com.neovisionaries.i18n.CountryCode::name)
                .orElseThrow(() -> new CountryCodeNotFoundException(countryCode, "ISO 3166-1 alpha-2"));
    }
}

File: service-provider-service/src/main/kotlin/com/wks/servicemarketplace/serviceproviderservice/core/CountryCode.kt
package com.wks.servicemarketplace.serviceproviderservice.core

import com.fasterxml.jackson.annotation.JsonValue
import com.wks.servicemarketplace.serviceproviderservice.core.exceptions.CountryCodeNotFoundException;

class CountryCode(code: String) {
    @JsonValue
    private val countryCode: String = com.neovisionaries.i18n.CountryCode.getByAlpha2Code(code)?.alpha2
            ?: throw CountryCodeNotFoundException(code, "ISO 3166-1 alpha-2")

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false
        return (other as CountryCode).countryCode == countryCode
    }

    override fun hashCode() = countryCode.hashCode()
    override fun toString() = countryCode
}